### PR TITLE
mozjs@1.8.5: fix compile issue

### PR DESCRIPTION
**Pull request by darmac**
According to https://github.com/apache/couchdb-pkg/tree/master/js/rpm/SOURCES
There is some patch for mozjs@1.8.5 to fix compile issue.

### CODE REVIEW

File: var/spack/repos/builtin/packages/mozjs/package.py [hunk 1]
```diff
@@ -31,15 +31,24 @@ class Mozjs(AutotoolsPackage):
     configure_directory = 'js/src'
     build_directory = 'js/src/spack-build'
 
-    patch('perl-bug.patch')
+    patch('perl-bug.patch', sha256='9f7d8502d85a4125e975a84cae11a8b34e696172d56f8ebc7ecf6d21fa3c30c9')
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/mozjs/package.py]: You do not have to include checksums for patches that are stored in the Spack repository

File: var/spack/repos/builtin/packages/mozjs/package.py [hunk 2]
```diff
@@ -31,15 +31,24 @@ class Mozjs(AutotoolsPackage):
...
             '--enable-threadsafe',  # enables support for multiple threads
             '--enable-system-ffi',
             '--with-system-zlib={0}'.format(spec['zlib'].prefix),
             '--with-system-nspr',
             '--with-nspr-prefix={0}'.format(spec['nspr'].prefix),
         ]
+        if spec.target.family == 'aarch64':
+            config_args.append('--host=aarch64-linux-gnu')
+        if spec.satisfies('@1.8.5'):
+            config_args.append('--disable-readline')
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/mozjs/package.py:56]: I think it's also reasonable to update the `depends_on` statement for `readline` like: `depends_on('readline', when='@17.0.0:')`

File: var/spack/repos/builtin/packages/mozjs/package.py [hunk 3]
```diff
@@ -31,15 +31,24 @@ class Mozjs(AutotoolsPackage):
     configure_directory = 'js/src'
     build_directory = 'js/src/spack-build'
 
-    patch('perl-bug.patch')
+    patch('perl-bug.patch', sha256='9f7d8502d85a4125e975a84cae11a8b34e696172d56f8ebc7ecf6d21fa3c30c9')
+    patch('Bug-638056-Avoid-The-cacheFlush-support-is-missing-o.patch',
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/mozjs/package.py:38]: Could you add a comment that links to the relevant bug report? It looks like that is https://bugzilla.mozilla.org/show_bug.cgi?id=638056

**Review by scheibelp — COMMENTED** [review]
I have a few (hopefully minor) requests

File: var/spack/repos/builtin/packages/mozjs/package.py (hunk 1 above)
Comment (darmac) [reply var/spack/repos/builtin/packages/mozjs/package.py]: fixed in https://github.com/spack/spack/pull/17594/commits/24e62fd4ca57b695cf80754394454d0e173dcae9

File: var/spack/repos/builtin/packages/mozjs/package.py (hunk 2 above)
Comment (darmac) [reply var/spack/repos/builtin/packages/mozjs/package.py:56]: fixed in https://github.com/spack/spack/pull/17594/commits/24e62fd4ca57b695cf80754394454d0e173dcae9

File: var/spack/repos/builtin/packages/mozjs/package.py (hunk 3 above)
Comment (darmac) [reply var/spack/repos/builtin/packages/mozjs/package.py:38]: fixed in https://github.com/spack/spack/pull/17594/commits/24e62fd4ca57b695cf80754394454d0e173dcae9

**Review by adamjstewart — APPROVED** [review]
I'm glad someone is using the `mozjs` package! I think I added it as a dependency of something else years ago, but I never got it working so I just contributed what I had at the time.

**Comment by darmac** [comment]
> I'm glad someone is using the `mozjs` package! I think I added it as a dependency of something else years ago, but I never got it working so I just contributed what I had at the time.

Infact, we just try to use `mozjs@1.8.5` on aarch64, the other versions still not available for aarch64 platform. XD

**Pull request merged**